plugins/backstage-plugin-coder/src/components/CoderAuthForm/CoderAuthForm.tsx

```diff
@@ -1,52 +1,30 @@
-import React, { type FC, type PropsWithChildren } from 'react';
-import { useCoderAuth } from '../CoderProvider';
-import { InfoCard } from '@backstage/core-components';
+import React from 'react';
+import { useInternalCoderAuth } from '../CoderProvider';
 import { CoderAuthDistrustedForm } from './CoderAuthDistrustedForm';
-import { makeStyles } from '@material-ui/core';
 import { CoderAuthLoadingState } from './CoderAuthLoadingState';
 import { CoderAuthInputForm } from './CoderAuthInputForm';
+import { CoderAuthSuccessStatus } from './CoderAuthSuccessStatus';
 
-const useStyles = makeStyles(theme => ({
-  cardContent: {
-    paddingTop: theme.spacing(5),
-    paddingBottom: theme.spacing(5),
-  },
-}));
+export type CoderAuthFormProps = Readonly<{
+  descriptionId?: string;
+}>;
 
-function CoderAuthCard({ children }: PropsWithChildren<unknown>) {
-  const styles = useStyles();
-  return (
-    <InfoCard title="Authenticate with Coder">
-      <div className={styles.cardContent}>{children}</div>
-    </InfoCard>
-  );
-}
-
-type WrapperProps = Readonly<
-  PropsWithChildren<{
-    type: 'card';
-  }>
->;
-
-export const CoderAuthWrapper = ({ children, type }: WrapperProps) => {
-  const auth = useCoderAuth();
-  if (auth.isAuthenticated) {
-    return <>{children}</>;
-  }
-
-  let Wrapper: FC<PropsWithChildren<unknown>>;
-  switch (type) {
-    case 'card': {
-      Wrapper = CoderAuthCard;
-      break;
-    }
-    default: {
-      assertExhaustion(type);
-    }
-  }
```

Comment:
This was my original attempt at trying to make `CoderAuthWrapper` be the one-stop-shop component for every auth component, but it just ended up being too wonky and restrictive:

1. Every component either has to accept zero props or the exact same props, no matter the use case
2. Up until now, I didn't have another component to try building with, so I never had a way to test if this design would actually hold up

Ended up making it so that there's a general `CoderAuthForm` component that offers the main functionality, and then it can be composed into more specialized components (`CoderAuthFormDialog` and `CoderAuthFormCardWrapper`)